econml/_cate_estimator.py

```diff
@@ -205,6 +215,54 @@ def ate(self, X=None, *, T0, T1):
         """
         return np.mean(self.effect(X=X, T0=T0, T1=T1), axis=0)
 
+    def cate_feature_names(self, feature_names=None):
+        """Public interface for getting feature names.
+
+        To be overriden by estimators that apply transformations the input features.
+
+        Parameters
+        ----------
+        feature_names: list of strings of length X.shape[1] or None
+            The names of the input features. If None and X is a dataframe, it defaults to the column names
+            from the dataframe.
+
+        Returns
+        -------
+        out_feature_names: list of strings or None
+            Returns feature names.
+        """
+        if feature_names is not None:
+            return feature_names
+        if hasattr(self, "_input_names"):
+            return self._input_names["feature_names"]
+        return None
+
+    def cate_output_names(self):
+        """
+        Public interface for getting output names.
+
+        Returns
+        -------
+        output_names: list of strings
+            Returns output names.
+        """
+        if hasattr(self, "_input_names"):
+            return self._input_names["output_names"]
+        return None
```

Comment:
Also they should be taking as input treatment_names and implement the override logic. This will save override logic from elsewhere. You are already doing this for cate_feature_names and I think we should do it here too.

Reply:
> These should be returning default names e.g. [y0, y1...] so that we uniformize the use of default names across our package.

This is already done, see above answer.

> Also they should be taking as input treatment_names and implement the override logic. This will save override logic from elsewhere. You are already doing this for cate_feature_names and I think we should do it here too.

Great suggestions, fixed!